challenges/functions.js

```diff
@@ -1,42 +1,54 @@
 // ==== Callbacks ====  
 
 /* Step 1: Create a higher-order function that accepts a callback
-  * Create a higher-order function named consume that can take 3 parameters.
-  * The first two parameters can accept any argument
-  * The last parameter accepts a callback 
-  * In the body of the function return the callback with the two parameters that you created
-*/
-
+ * Create a higher-order function named consume that can take 3 parameters.
+ * The first two parameters can accept any argument
+ * The last parameter accepts a callback 
+ * In the body of the function return the callback with the two parameters that you created
+ */
+function consume(a1, a2, cb) {
+    return cb(a1, a2);
+}
 
 /* Step 2: Create several functions to callback with consume();
-  * Create a function named add that returns the sum of two numbers
-  * Create a function named multiply that returns the product of two numbers 
-  * Create a function named greeting that accepts a first and last name and returns "Hello first-name last-name, nice to meet you!"
-*/
+ * Create a function named add that returns the sum of two numbers
+ * Create a function named multiply that returns the product of two numbers 
+ * Create a function named greeting that accepts a first and last name and returns "Hello first-name last-name, nice to meet you!"
+ */
+function add(num1, num2) {
+    return num1 + num2;
+}
 
+function multiply(num1, num2) {
+    return num1 * num2;
+}
+
+function greeting(firstname, lastname) {
+    return `Hello ${firstname} ${lastname}, nice to meet you!`;
+}
 
 /* Step 3: Check your work by un-commenting the following calls to consume(): */
-// consume(2,2,add); // 4
-// consume(10,16,multiply); // 160
-// consume("Mary","Poppins", greeting); // Hello Mary Poppins, nice to meet you!
+console.log(consume(2, 2, add)); // 4
+console.log(consume(10, 16, multiply)); // 160
+console.log(consume("Mary", "Poppins", greeting)); // Hello Mary Poppins, nice to meet you!
 
 
 // ==== Closures ==== 
 
 // Explain in your own words why `nestedfunction()` can access the variable `internal`.
 
-// Explanation: 
+// Explanation: The variable 'internal' is first defined in the local scope of myFunction. After that, the code has a nested function that logs 'internal.' We then have a closure that gives 'nestedfunction()' access to 'internal' because it is able to reach outside its scope chain to the outer scope as a closure.
 
 
 const external = "I'm outside the function";
 
 function myFunction() {
-  console.log(external);
-  const internal = "Hello! I'm inside myFunction!";
+    console.log(external);
+    const internal = "Hello! I'm inside myFunction!";
 
-  function nestedFunction() {
-    console.log(internal);
-  };
-  nestedFunction();
+    function nestedFunction() {
+        console.log(internal);
+    };
+    nestedFunction();
 }
```

Comment:
the magic of closures is being able to use a function that stores a variable outside of it, and even if that varible doesn't exist anymore in myFunction(), the nestedFunction still is utilizing it.